DR_0011_React-for-paginated-views.md

```diff
@@ -0,0 +1,194 @@
+      {"label": "Type", "field": "type", "sortable": true}
+    ],
+    "row_component": Row
+  }
+  ReactDOM.render(
+    React.createElement(PaginatedTable, props),
+    document.getElementById('paginated-table')
+  );
+</script>
+```
+
+A very basic implementation can be found in [branch `react-party`](https://github.com/Cadasta/cadasta-platform/compare/react-party).
+
+### Database changes
+
+N/A
+
+### Security considerations
+
+N/A
+
+### Scalability
+
+When implemented, this approach will improve scalability of the platform because smaller sets will be process in the database and the application server. 
+
+### Maintainability
+
+#### Package management
+
+We will use [yarn](https://yarnpkg.com) for package management. Yarn provides some advantages over npm:
+
+- It is faster.
+- Builds fail less often when a package can't be installed. It retries the failing request instead. 
+- It is deterministic. The same libraries are installed across all systems; with npm you sometimes had different versions of the same library installed depending on the install order and the whether other libraries depend on the same library but a different version. 
+
+#### Task runner
+
+We will use [Grunt](https://gruntjs.com/) as our task runner. Grunt will be used to create builds, run tests and lint the code.  
+
+#### Builds
+
+The build compiles from `jsx` to JavaScript and minifies that code. The build script should be run when the test server is started in the dev VM (including a watch script so sources are re-built whenever a file is changed) and when deploying to production. 
+
+We will run individual builds for each component to keep the size of resources small. For example, there will be one resource for paginated pages, another one for WMS, etc.
+
+The command to create builds will be `grunt build`.
+
+#### Tests
+
+We will start to unit-test front-end components. We are going to use [Jest](https://facebook.github.io/jest/) for our test framework and [Enzyme](http://airbnb.io/enzyme/) to make it easier to assert and traverse React components.
+
+The Travis build will have an additional test environment that runs all unit tests for client-side components. 
+
+The command to run tests will be `grunt test`.
+
+#### Code linting
+
+We will the client-side code using [eslint](http://eslint.org/) to ensure a minimum standard code quality. Linting will be run as a separate Grunt task.
+
+The Travis build will have an additional test environment that runs eslint on all client-side components.
+
+The command to run code linting will be `grunt lint`.
+
+#### Monitoring dependencies 
+
+We use [Greenkeeper](https://greenkeeper.io/) to monitor the status of npm dependencies.  It works similar to requires.io, whenever dependencies are upgraded we recieve a pull request ([Here's an example](https://github.com/oliverroick/Leaflet.Deflate/pull/12/files)).
+
+### Costs considerations
+
+An organization account for Greenkeeper is $25/month.
```

Comment:
Greenkeeper is [free for public repos](https://greenkeeper.io/#pricing), so we shouldn't require a paid organization account.

Reply:
Right, I changed that.